### ISSUE TITLE
Review the basic session description

### ISSUE THREAD

**Issue by simonschllng** [ocpi/ocpi#45]
Please find the initial part of the description for the session object here: https://github.com/ocpi/ocpi/blob/master/bo_sessions.md

I'm quite confident with the description, the possibilities of the UML diagram however are limiting here.

**Comment by simonphilips** [comment]
Perhaps the definitions can be better described as following:

![session](https://cloud.githubusercontent.com/assets/3263809/9989373/2764912a-6058-11e5-9f01-9639f63a2f74.png)

We can first describe the generic Session type and then document the EMSPSession and CPOSession specific fields. In this case the endpoints field will be different and even have a slightly different definition (different options for eMSP <> CPO).

The flow diagram is then:

![sessionflow](https://cloud.githubusercontent.com/assets/3263809/9989573/de1fb70e-6059-11e5-85e3-7db630776e8b.png)

At least this gives a clear picture of what kind of object each party has to send and what object to expect for each request.

**Comment by simonphilips** [comment]
Similarly we could use the following for the credentials endpoint. The object would look like this:

![credentials](https://cloud.githubusercontent.com/assets/3263809/9990071/499ecdea-605e-11e5-97b7-49319fbc3537.png)

The flow would then be described by this diagram:

![credentialsflow1](https://cloud.githubusercontent.com/assets/3263809/9990000/acf36398-605d-11e5-81ce-ef5a6599b340.png)

And of course, in the credentials case, the reverse flow is also possible:

![credentialsflow2](https://cloud.githubusercontent.com/assets/3263809/9990001/ae8be478-605d-11e5-8479-9f9406b4a223.png)

Again, each response is very clearly typed so everyone knows what to send and what to receive. I think this is a much better approach than what we currently have. What do you think?

**Comment by simonschllng** [comment]
This view is much more clear. However, in the initial description I struggled with the fact that the session can be initiated by both, operator and provider. By having a virtual object in the flow diagram I tried to describe that there can be a ongoing session, even if the other party did not yet receive that info.
From your flow diagram, one could easily understand that the session object lives in the eMSP system and can only be initiated through a POST request by the operator. In cases where the eMSP is not responding, the operator would still start the session. This has to be clear.
Any suggestions how to cover that?

**Comment by simonphilips** [comment]
I think the initiation by the eMSP should be done differently. I would prefer to see that as part of the start command for an EVSE: you start an EVSE => you trigger a POST session from the CPO to the eMSP.

**Comment by simonschllng** [comment]
This would mean, that the CPO is the master over the session. The session object lives in the CPO system and is copied – or inherited – to the eMSP system. Then I would not include any start or stop mechanism into the session. The operator handles the session and keeps it updated. The provider pull it or gets it pushed. However, to stop or alter the charging process the provider has to call a EVSE-related method.

#### Questions:
 * Did you already discuss the start, stop or control mechanism? (Provider controls a operators charging station)
 * @simonphilips: Can you share the source of your diagrams above?

I'm currently merging the work from [Branch 20150828123425_sessions_chapter](https://github.com/ocpi/ocpi/blob/20150828123425_sessions_chapter/sessions.md).

**Comment by simonphilips** [comment]
> This would mean, that the CPO is the master over the session. The session object lives in the CPO system and is copied – or inherited – to the eMSP system. 

Sort of, yes. The CPO controls the session. The CPO is the only one who can start a session. It is also the only one who can truly stop a session.

If the eMSP wants to stop a session, it should probably use some command which will let the CPO know it wants to stop (that part is not discussed yet). It is then up to the CPO to initiate the stop sequence and eventually issue a DELETE request to the eMSP to indicate the session is in fact stopped.

So to summarize:

**CPO:**
- force start: POST /emsp/sessions
- force stop: DELETE /emsp/sessions/{id}

**eMSP:**
- request start: POST /cpo/evses/{id}/start (or something similar, this is not decided yet)
- request stop: either through POST /cpo/evses/{id}/stop or DELETE /cpo/sessions/{id} (or something similar, this is not decided yet)
